### ISSUE TITLE
?:= for handling optional attributes

### ISSUE THREAD

**Issue by baank** [shadaj/slinky#274]
Proposal to add a ?:= method which will take an Option as the right hand argument.
If the value of the Option is None then the entire attribute is removed.

For example:
```
@react class MyButton {
  case class Props(onClick: Option[() => Unit] = None)
  
  def render = 
    button(onClick ?:= props.onClick)
}
```

```
MyButton(onClick = None)
```

For example if I use the above then would expect to have a button without a click handler. Currently I have to duplicate a lot of code to handle this.

**Comment by shadaj** [comment]
This makes a lot of sense, there's really no good way to handle situations like this right now. I'll milestone this for 0.7.0 (hoping to get 0.6.2 out as a quick patch). If you're interested in contributing this, that would be great!

**Comment by baank** [comment]
@shadaj .. Sounds like a good one to learn the codebase so will try and submit a PR in the next few days.

**Comment by baank** [comment]
I have fixed this in PR #275.

I decided to just overload the := syntax but happy to leave the choice up to you.
